### DISCUSSION TITLE
Dynamic tailwind class does not apply to the element (nuxt)

### DISCUSSION THREAD

**Discussion by m0k99** [nuxt/nuxt#25826] [Questions] [answered]
I have the `input.vue` component and I want to get the class config from the `input.ts` file I do this truly but I import this file in my `input.vue` file and use it in `:class` my class config type is tailwind and when I set them on input class they does not apply 
my code:

```ts
//input.ts
export default {
    base:'relative block w-full disabled:cursor-not-allowed disabled:opacity-75 focus:outline-none border-0',
    rounded:'rounded-full',
    placeholder: 'placeholder-gray-400 dark:placeholder-gray-500',
    padding:'py-3.5 px-2.5'
}

```

```vue
//input.vue
<template>
  <div>
    <input
      :class="className"
    >
  </div>
</template>

<script lang="ts">
import {defineComponent} from 'vue'
import {default as inputUi} from "../../ui.config/input.ts";
import {twJoin, twMerge} from "tailwind-merge";
export default defineComponent({
  setup(props) {
    const className = ref('')
    className.value = twMerge(twJoin(
      inputUi.base,
      inputUi.rounded,
      inputUi.placeholder,
      inputUi.padding,
    ))
    return {
      className
    }
  }
})
</script>
```
how do I fix it until my **tailwind** class apply

**Comment by TheAlexLichter** [comment]
Hey! Could you kindly provide a [reproduction](https://nuxt.com/docs/community/reporting-bugs/#create-a-minimal-reproduction)? 🙏

Also, I don't think this needs to be a ref, it can be a plain variable ☺️

**Reply by m0k99** [reply]
thanks, @manniL  for your answer 🙏
In my code, I have a `classInput` prop that `computed` with twMerge with twJoin class so I use `ref` but for summarising the question I remove computed and `classInput` props.

**Comment by m0k99** [comment, marked as answer]
After hours and testing the paths and looking at the [tailwind document](tailwindcss.com/docs/content-configuration#configuring-source-paths)  I found that all the files we need to use the class must be scanned by tailwind CSS, so it must be defined in `content` in the tailwind configuration. So I add `/ui.config/**/*.{js,ts}.`


```ts
import type {Config} from 'tailwindcss'

export default {
  content: [
    `./components/**/*.{vue,js,ts}`,
    `./layouts/**/*.vue`,
    `./pages/**/*.vue`,
    `./composables/**/*.{js,ts}`,
    `./plugins/**/*.{js,ts}`,
    `./utils/**/*.{js,ts}`,
    `./ui.config/**/*.{js,ts}`,
    `./App.{js,ts,vue}`,
    `./app.{js,ts,vue}`,
    `./Error.{js,ts,vue}`,
    `./error.{js,ts,vue}`,
    `./app.config.{js,ts}`
  ],
} satisfies Config

```
